keel-core/src/main/kotlin/com/netflix/spinnaker/keel/exceptions/NonexistentArtifactReferenceException.kt

```diff
@@ -0,0 +1,7 @@
+package com.netflix.spinnaker.keel.exceptions
+
+class NonexistentArtifactReferenceException(
+  reference: String
+) : ValidationException(
+  "Config uses an artifact reference that does not correspond to any artifacts: $reference."
```

Comment:
It would be nice to also show existing references in the config to make the mistake easy to spot

Reply:
@emjburns Do you mean augment the error message to include the list of artifact references that are properly defined?

Reply:
yes! exaclty.

Reply:
"Config uses an artifact reference ($reference) that does not correspond to any artifacts. Valid references are: $validReferences." ?

Reply:
Addressed in b5d56502ade93d447134e5eea90406a5f3559183